hugolib/site_new.go

```diff
@@ -558,3 +559,9 @@ func (s *Site) AllPages() page.Pages {
 func (s *Site) AllRegularPages() page.Pages {
 	return s.h.RegularPages()
 }
+
+// Translate returns a translated string for id.
+func (s *Site) Translate(id string, args ...any) (string, error) {
+	ctx := context.Background()
```

Comment:
Add `ctx` as the first arg in `Translate`. It will be "auto injected".

That said. I don't remember the original issue, but looking at the below I would say that this methods belongs on the `Language` struct.

You would need to say `.Site.Language.Translate` ... which is a little longer ... but cleaner.